### PR TITLE
#9 admin controller & spec

### PR DESCRIPTION
**Pull request by wozane**
- connected to #9 
- view will be completed by @basiacierocka

### CODE REVIEW

File: app/controllers/admin/recipes_controller.rb [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+  def show
+    @recipe = Recipe.find(params[:id])
+  end
+
+  def new
+    @recipe = Recipe.new
+  end
+
+  def create
+    @recipe = Recipe.new(recipie_params)
```
Comment (basiacierocka) [inline app/controllers/admin/recipes_controller.rb]: shouldn't be(recipe_params) at 17?

Comment (wozane) [inline app/controllers/admin/recipes_controller.rb]: 👍

**Comment by CallMeSandra** [comment]
I think you should delete all specs except recipes_controller_spec.rb and ingredients_controller_spec.rb; they're empty anyway

File: app/controllers/admin/ingredients_controller.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
+class Admin::IngredientsController < ApplicationController
+  http_basic_authenticate_with name: "admin", password: "1234"
```
Comment (gotar) [inline app/controllers/admin/ingredients_controller.rb]: you should take those info from ENV[''], and set in production using `heroku config:set`

in test, dev you should take from `config/application.yaml` or so

File: app/controllers/admin/ingredients_controller.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+  end
+
+  def new
+    @Ingredient = Ingredient.new
+  end
+
+  def create
+    @ingredient = Ingredient.new(ingredient_params)
+
+    @ingredient.save
```
Comment (gotar) [inline app/controllers/admin/ingredients_controller.rb]:
``` ruby
if @ingredient.save
  redirect_to @ingredient, notice: "Ingredient added"
else
  render action :new
end
```

File: app/controllers/admin/recipes_controller.rb [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def update
+    @recipe = Recipe.find(params[:id])
+
+    if @recipe.update(params[:id])
+      redirect_to @recipe
+    else
+      render 'edit' #this is a connection to the view
+    end
```
Comment (gotar) [inline app/controllers/admin/recipes_controller.rb:33]: something similar should be in `new`

File: app/controllers/admin/recipes_controller.rb [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  end
+
+  def destroy
+    @recipe = Recipe.find(params[:id])
+    @recipe.destroy
+
+    redirect_to admin/recipe_path
```
Comment (gotar) [inline app/controllers/admin/recipes_controller.rb]: as a string "admin/recipe_path" or better admin_recipe_path (check correct one in `rake routes`)

File: app/helpers/admin/ingredients_helper.rb [hunk 6]
```diff
@@ -0,0 +1,2 @@
+module Admin::IngredientsHelper
+end
```
Comment (gotar) [inline app/helpers/admin/ingredients_helper.rb]: you can drop this file

File: app/helpers/admin/recipes_helper.rb [hunk 7]
```diff
@@ -0,0 +1,2 @@
+module Admin::RecipesHelper
+end
```
Comment (gotar) [inline app/helpers/admin/recipes_helper.rb]: you can drop this file

File: app/controllers/admin/ingredients_controller.rb [hunk 8]
```diff
@@ -0,0 +1,43 @@
...
+
+    if @ingredient.save
+      redirect_to @ingredient, notice: "Ingredient saved"
+    else
+      render 'new'
+    end
+  end
+
+    def update
+    @ingredient = Ingredient.find(params[:id])
```
Comment (gotar) [inline app/controllers/admin/ingredients_controller.rb:23]: wrong indentation

**Comment by gotar** [comment]
ok looks good you can merge

File: app/controllers/admin/ingredients_controller.rb (hunk 8 above)
Comment (wozane) [inline app/controllers/admin/ingredients_controller.rb:23]: fixed

**Pull request merged**